### PR TITLE
x/cqlbuilder: Add implementation of a query builder

### PR DESCRIPTION
**Pull request by AlexisMontagne**
### What does this PR do?

C* is the second most used database at Upfluence, from the experience learnt of using upfluence/sql/x/sqlbuilder the idea is to provide similar DSL to create easy to build / composable request to a cassandra cluster.

### What are the observable changes?

Too much to mention here.

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Properly labeled

### CODE REVIEW

File: x/cqlbuilder/query_builder.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package cqlbuilder
+
+import (
+	"github.com/upfluence/cql"
+)
```
Comment (tgallice) [inline x/cqlbuilder/query_builder.go:5]:
```suggestion
import "github.com/upfluence/cql"
```
:p

File: x/cqlbuilder/query_builder_test.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/upfluence/cql"
+	"github.com/upfluence/cql/cqltest"
+	"github.com/upfluence/cql/x/migration"
+)
```
Comment (tgallice) [inline x/cqlbuilder/query_builder_test.go:11]:
```suggestion
import (
	"context"
	"testing"

	"github.com/stretchr/testify/assert"
	
	"github.com/upfluence/cql"
	"github.com/upfluence/cql/cqltest"
	"github.com/upfluence/cql/x/migration"
)
```

**Comment by tgallice** [comment]
Otherwise, my very small comments, LGTM

File: x/cqlbuilder/execer.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	return cs.sc.ScanCAS(vs...)
+}
+
+type Execer interface {
+	Exec(context.Context, map[string]interface{}) error
+	ExecCAS(context.Context, map[string]interface{}) CASScanner
+}
+
+type execer struct {
```
Comment (jlevesy) [inline x/cqlbuilder/execer.go:45]: Cassandra noob here,

What's the difference between `Exec` and `ExecCAS` ? Expecting a scanner to deal with some output ? If so what does CAS stands for ?

File: x/cqlbuilder/query_builder.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	buildQuery(map[string]interface{}) (string, []interface{}, error)
+}
+
+type CASStatement interface {
+	statement
+
+	casScanKeys() []string
+}
+
+type InsertExecer struct {
```
Comment (jlevesy) [inline x/cqlbuilder/query_builder.go:51]: I'm not sure to understand the point of those `Insert|Update|Delete` execers ? Why not returning an `execer` directly ?

File: x/cqlbuilder/query_builder_test.go [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+			},
+		)
+
+		ok, _, _, err = execCAS(de, "foo", "")
+		assert.True(t, ok)
+		assert.NoError(t, err)
+	})
+}
+
+func TestEC(t *testing.T) {
```
Comment (jlevesy) [inline x/cqlbuilder/query_builder_test.go:111]: What does EC stands for ? What does this test tests ?

**Review by jlevesy — COMMENTED** [review]
I only read it partially.

I would be happy to get a pair review session with you, that would be more efficient IMO. I'm not familliar enough with cassandra to be relevant here.

**Pull request merged**